app/assets/javascripts/discourse/app/static/prosemirror/extensions/link-toolbar.js

```diff
@@ -0,0 +1,319 @@
+                            );
+                            dispatch(tr);
+                            view.focus();
+                          }
+                        },
+                      },
+                    },
+                  });
+                },
+
+                copyLink: async () => {
+                  await clipboardCopy(linkState.href);
+                  getContext().toasts.success({
+                    duration: "short",
+                    data: {
+                      message: i18n("composer.link_toolbar.link_copied"),
+                    },
+                  });
+                },
+
+                unlinkText: () => {
+                  const range = view.state.selection.empty
+                    ? linkState.range
+                    : view.state.selection;
+                  if (range) {
+                    const { state, dispatch } = view;
+                    dispatch(
+                      state.tr.removeMark(
+                        range.from,
+                        range.to,
+                        state.schema.marks.link
+                      )
+                    );
+                    view.focus();
+                  }
+                },
+
+                canVisitLink: () => {
+                  return utils
+                    ? !!utils.getLinkify().matchAtStart(linkState.href)
+                    : false;
+                },
+
+                getHref: () => linkState.href,
+
+                canUnlink: () => !AUTO_LINKS.includes(linkState.markup),
+              };
+
+              linkToolbar = new LinkToolbar(handlers);
+              linkToolbar.rovingButtonBar = (event) => {
+                if (event.key === "Tab") {
+                  event.preventDefault();
+                  view.focus();
+                  return false;
+                }
+                return rovingButtonBar(event);
+              };
+            } else {
+              linkState = attrs;
+            }
+
+            if (getContext().capabilities.viewport.sm) {
+              const element = view.domAtPos(attrs.head).node;
+              const trigger =
+                element.nodeType === Node.TEXT_NODE
+                  ? element.parentElement
+                  : element;
+
+              trigger.getBoundingClientRect = () => {
+                if (!view.docView) {
+                  return {};
+                }
+
+                const { left, top } = view.coordsAtPos(attrs.head);
+                return {
+                  left,
+                  top: top + MENU_OFFSET,
+                  width: 0,
+                  height: 0,
+                };
+              };
+
+              if (menuInstance) {
+                if (shouldUpdateMenu) {
+                  menuInstance.trigger = trigger;
+                  updatePosition(
+                    menuInstance.trigger,
+                    menuInstance.content,
+                    {}
+                  );
```

Comment:
I wonder if we shouldn't haven't this function on the menu instance itself, so we don't have to import float-kit internals